sdk/inc/azure/core/_az_cfg.h

```diff
@@ -88,4 +88,31 @@
 // Get the number of elements in an array
 #define _az_COUNTOF(array) (sizeof(array) / sizeof((array)[0]))
 
+/**
+ * @brief Deprecate functions.
+ *
+ */
+#ifdef __has_c_attribute
+#if __has_c_attribute(deprecated)
+#define AZ_DEPRECATED [[deprecated]]
+#endif // __has_c_attribute(deprecated)
+#endif // __has_c_attribute
+
+#ifndef AZ_DEPRECATED
+#if defined(_MSC_VER)
+#define AZ_DEPRECATED __declspec(deprecated)
+#elif defined(__GNUC__) // !defined(_MSC_VER)
+#define AZ_DEPRECATED __attribute__((deprecated))
+#endif // defined(_MSC_VER)
+#endif // AZ_DEPRECATED
+
+#ifdef _MSC_VER
+#define AZ_PUSH_IGNORE_DEPRECATIONS _Pragma("warning(push)") _Pragma("warning(disable:4996)")
+#define AZ_POP_WARNINGS _Pragma("warning(pop)")
+#else // !_MSC_VER
+#define AZ_PUSH_IGNORE_DEPRECATIONS \
+  _Pragma("GCC diagnostic push") _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
+#define AZ_POP_WARNINGS _Pragma("GCC diagnostic pop")
+#endif // _MSC_VER
```

Comment:
I just noticed that these are prefixed with `AZ_` which are meant to indicate for public use (especially when accessible from a header).

These helper macros are only used in tests and as a stop-gap in samples. Ideally we shouldn't have them at all, as there isn't substantive usage of it.

But, if we want to keep them, they should be for private use within the SDK only, which is indicated with the prefix `_az_`.

So, either we remove them, or rename them as:
`_az_POP_WARNINGS`

@CIPop let me know and I can make the fix, if you need me to, before we release 1.5.0 GA today.

Reply:
Yes, these are intended for public use (they are used in our samples). This was intentional, to allow existing applications to work around the deprecation until the API will be removed in V2. No changes required.